src/html.c

```diff
@@ -206,28 +206,57 @@ search_links (const gchar* data, gint linkType)
 	return result;
 }
 
-gchar *
-html_auto_discover_feed (const gchar* data, const gchar *baseUri)
+static void
+html_auto_discover_collect_links (xmlNodePtr match, gpointer user_data)
 {
-	gchar		*res, *tmp;
-	const gchar	*baseU;
+	GSList **links = (GSList **)user_data;
+	gchar *link = xml_get_attribute (match, "href");
+	if(link)
+		*links = g_slist_append (*links, link);
+}
 
-	baseU = search_tag_link(data, "base", NULL);
-	if (!baseU)
-		baseU = baseUri;
+GSList *
+html_auto_discover_feed (const gchar* data, const gchar *defaultBaseUri)
+{
+	GSList		*iter, *links = NULL;
+	gchar		*baseUri;
```

Comment:
Found a source of crashes/garbage source : this should be initialized to NULL or it will contain garbage and not be null in the if below.

Reply:
Thanks for spotting this! It is amazing how useless Githubs CodeQL scanner is to not find this uninitialized variable. Maybe I misconfigured it, but so far it found nothing of relevance.